src/properties/module/index.js

```diff
@@ -1,36 +1,86 @@
 import Joi from 'joi'
+import { oneLine } from 'common-tags'
 
-export const CONDITION_MESSAGE =
-  'may be a RegExp (tested against absolute path), ' +
-  'a string containing the absolute path, a function(absPath): bool, ' +
-  'or an array of one of these combined with “and”.'
+export const CONDITION_MESSAGE = oneLine`
+  may be a RegExp (tested against absolute path),
+  a string containing the absolute path, a function(absPath): bool,
+  or an array of one of these combined with “and”.`
 
-export const LOADERS_QUERY_MESSAGE =
-  'You can only pass the \`query\` property when you specify your ' +
-  'loader with the singular \`loader\` property'
+export const LOADERS_QUERY_MESSAGE = oneLine`
+  You can only pass the \`query\` property when you specify your
+  loader with the singular \`loader\` property`
+
+const LOADER_INCLUDE_OR_EXCLUDE_MESSAGE = oneLine`!!
+  Please supply \`include\` or/and \`exclude\` to narrow down which
+  files will be processed by this loader.
+  Otherwise it's too easy
+  to process too many files, for example your \`node_modules\` (loader-enforce-include-or-exclude).
+`
+
+const LOADER_INCLUDE_REQUIRED = oneLine`
+  is required. You should use \`include\` to specify which files should be processed
+  by this loader. (loader-prefer-include)
+`
+
+const LOADER_EXCLUDE_FORBIDDEN = oneLine`
+  should not be used. Reason: \`exclude\` makes it easy to match too many files,
+  which might inadvertently slow your build down. Use \`include\` instead. (loader-prefer-include)
+`
 
 const conditionSchema = Joi.array().items([
   Joi.string(),
   Joi.object().type(RegExp),
   Joi.func().arity(1),
 ]).single().options({ language: { array: { includesSingle: CONDITION_MESSAGE } } })
 
-const loaderSchema = Joi.object({
-  test: conditionSchema.required(),
-  exclude: conditionSchema,
-  include: conditionSchema,
-  loader: Joi.string(),
-  query: Joi.object(),
-  loaders: Joi.array().items(Joi.string()),
-})
-  .xor('loaders', 'loader')
-  .nand('loaders', 'query').options({ language: { object: { nand: LOADERS_QUERY_MESSAGE } } })
-
-const loadersSchema = Joi.array().items(loaderSchema)
-
-export default Joi.object({
-  loaders: loadersSchema.required(),
-  preLoaders: loadersSchema,
-  postLoaders: loadersSchema,
-  noParse: Joi.array(Joi.object().type(RegExp)).single(),
-})
+const loaderSchemaFn = ({ rules }) => {
+  let rule = Joi.object({
+    test: conditionSchema.required(),
+    exclude: conditionSchema,
+    include: conditionSchema,
+    loader: Joi.string(),
+    query: Joi.object(),
+    loaders: Joi.array().items(Joi.string()),
+  })
+    .xor('loaders', 'loader')
+    .nand('loaders', 'query')
+    .options({ language: { object: { nand: LOADERS_QUERY_MESSAGE } } })
+
+  if (rules['loader-enforce-include-or-exclude']) {
```

Comment:
So I guess this is how the general pattern of applying rules to the schema will look like. If you come up with a better pattern, please don't hesitate. :)